src/gas-price/gas-price.ts

```diff
@@ -141,37 +122,38 @@ export const getRecommendedGasPrice = (chainId: string, providerName: string, sp
       multiplier: scalingMultiplier,
       pendingPeriod,
     });
-    return multiplyBigNumber(latestGasPrice, scalingMultiplier);
+    gasPriceToUse = multiplyBigNumber(latestGasPrice, scalingMultiplier);
   }
 
   // Check that there are enough entries in the stored gas prices to determine whether to use sanitization or not
-  // Calculate the minimum timestamp that should be within the 90% of the sanitizationSamplingWindow
+  // Calculate the minimum timestamp that should be within the 90% of the sanitizationSamplingWindow.
   const minTimestamp = Math.floor(Date.now() / 1000) - 0.9 * sanitizationSamplingWindow;
-
-  // Check if there are entries with a timestamp older than at least 90% of the sanitizationSamplingWindow
+  // Check if there are entries with a timestamp older than at least 90% of the sanitizationSamplingWindow.
   const hasSufficientSanitizationData = gasPrices.some((gasPrice) => gasPrice.timestamp <= minTimestamp);
+  // Get the configured gas price cap for sanitization.
+  const sanitizationGasPriceCap = multiplyBigNumber(
+    getPercentile(
+      sanitizationPercentile,
+      gasPrices.map((gasPrice) => gasPrice.price)
+    )!,
+    sanitizationMultiplier
+  );
 
-  // Get the configured percentile of historical gas prices
-  const percentileGasPrice = getPercentile(
-    sanitizationPercentile,
-    gasPrices.map((gasPrice) => gasPrice.price)
-  )!;
-  // Log a warning if there is not enough historical data to sanitize the gas price but the price could be sanitized
-  if (!hasSufficientSanitizationData && latestGasPrice > percentileGasPrice) {
+  // Log a warning if there is not enough historical data to sanitize the gas price but the price could be sanitized.
+  if (!hasSufficientSanitizationData && gasPriceToUse > sanitizationGasPriceCap) {
     logger.warn('Gas price could be sanitized but there is not enough historical data.', {
-      gasPrice: latestGasPrice.toString(),
-      percentileGasPrice: percentileGasPrice.toString(),
+      gasPrice: gasPriceToUse.toString(),
+      sanitizationGasPriceCap: sanitizationGasPriceCap.toString(),
     });
   }
-
-  // If necessary, sanitize the gas price and log a warning because this should not happen under normal circumstances
-  if (hasSufficientSanitizationData && latestGasPrice > percentileGasPrice) {
+  // If necessary, sanitize the gas price and log a warning because this should not happen under normal circumstances.
+  if (hasSufficientSanitizationData && gasPriceToUse > sanitizationGasPriceCap) {
     logger.warn('Sanitizing gas price.', {
-      gasPrice: latestGasPrice.toString(),
-      percentileGasPrice: percentileGasPrice.toString(),
+      gasPrice: gasPriceToUse.toString(),
+      sanitizationGasPriceCap: sanitizationGasPriceCap.toString(),
     });
-    return multiplyBigNumber(percentileGasPrice, recommendedGasPriceMultiplier);
+    return sanitizationGasPriceCap;
```

Comment:
Probably we should be more generous while configuring `sanitizationMultiplier` because now it is also capping the result from scaling.

Reply:
Yeah. Also note that it might be a good idea to set `recommendedGasPriceMultiplier` to `1` because now it's going to be the latest gas price.

This is something that should be also changed in the example files to make it more clear.